Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceUpdateExecutor.cpp

```diff
@@ -110,17 +119,19 @@ namespace AzToolsFramework
             {
                 if (instance != instanceToExcludePtr)
                 {
-                    m_instancesUpdateQueue.emplace_back(instance);
+                    AddInstanceToQueue(instance);
                 }
             }
         }
 
-        void InstanceUpdateExecutor::RemoveTemplateInstanceFromQueue(const Instance* instance)
+        void InstanceUpdateExecutor::RemoveTemplateInstanceFromQueue(Instance* instance)
         {
             AZStd::erase_if(m_instancesUpdateQueue, [instance](Instance* entry)
             {
                 return entry == instance;
             });
+
+            m_uniqueInstancesForPropagation.erase(instance);
```

Comment:
You could use the return value of erase here to determine if the element was in the set - and skip the erase_if if the item was not in the set.